### ISSUE TITLE
Regression: shadow of desktop items are gone.

### ISSUE THREAD

**Issue by PCMan** [lxde/pcmanfm-qt#304]
This is what the desktop items should look like.
![shadow](https://cloud.githubusercontent.com/assets/46543/12741488/c123e252-c9b7-11e5-885d-935bce3368d2.png)
This is the current appearance.
![no_shadow](https://cloud.githubusercontent.com/assets/46543/12741489/c128b0de-c9b7-11e5-901a-d71cacafe77e.png)
The bug is only triggered on the initial startup of pcmanfm-qt.
After the layout of desktop items is changed, the shadows appear.
In addition, if you open the desktop preference dialog and apply the settings again, the shadows also appear.
I suspect this is related to recent item delegate changes.
@tsujan : Would you please take a look at this issue?
Thanks a lot.

**Comment by tsujan** [comment]
Thanks!

in `desktopitemdelegate.cpp` -> `if((opt.state & QStyle::State_Selected) && opt.widget)`, the condition `opt.widget` wasn't there before; instead it was so:
````
if(opt.state & QStyle::State_Selected) {
  QPalette palette = widget->palette();
  ...
}
````
which wasn't correct either. I guess I'd worked with an older version and then applied the patch to the one with `opt.widget`, without noticing it. I'll make a PR soon.

**Comment by tsujan** [comment]
@PCMan
I couldn't reproduce the issue here, so I can't know the PR fixes it, but the above-mentioned condition wasn't correct either.

**Comment by PCMan** [comment]
it works here. If later the bug appears again, I'll call you. lol

**Comment by tsujan** [comment]
@PCMan
I can't see how https://github.com/lxde/pcmanfm-qt/pull/306 could have fixed the problem you encountered (although that commit corrected a mistake). Accidentally I succeeded in producing shadowless texts by changing the position of the panel or adding a new one. The shadow returned with layout change, as you had described.

I'll try to find the real cause.

**Comment by PCMan** [comment]
Yes, the bug happened again later. So it seems to be an accident that the bug disappear previously. I can reproduce it now. So let's reopen the issue.

**Comment by tsujan** [comment]
@PCMan
I made some tests and found out that the text shadow color is reset to `(0,0,0,0)` when `virtualGeometryChanged()` is emitted. I may be lost but this is my theory:

Before `FolderView::setMargins()` was introduced, nothing happened. But this method calls `setGridSize()`, which causes the items to be laid out again. So, I think that the resetting of shadow has something to do with the duplication of `FolderItemDelegate` as `DesktopItemDelegate`.

I included a simple workaround in https://github.com/lxde/pcmanfm-qt/pull/303 (at the end of `DesktopWindow::removeBottomGap()`). If the whole PR isn't going to be merged, I could make a separate PR with just that fix.
